mocker.go

```diff
@@ -0,0 +1,108 @@
+package testing
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/juju/loggo"
+)
+
+// NewCallMocker returns a CallMocker which will log calls and results
+// utilizing the given logger.
+func NewCallMocker(logger loggo.Logger) *CallMocker {
+	return &CallMocker{
+		logger:  logger,
+		results: make(map[string][]*callMockReturner),
+	}
+}
+
+// CallMocker is a tool which allows tests to dynamically specify
+// results for a given set of input parameters.
+type CallMocker struct {
+	Stub
+
+	logger  loggo.Logger
```

Comment:
It seems like it would unnecessarily couple this type to gocheck. I wish gc.C satisified some kind of logger interface so that we could just pass it in.